bootstrap/src/lib/cache-warmer/config.ts

```diff
@@ -0,0 +1,35 @@
+import objectHash from 'object-hash'
+
+export interface Config {
+  /**
+   * The interval in milliseconds which the warm-up engine will execute
+   * the underlying external adapter to update its cache
+   */
+  warmupInterval: number
+  /**
+   * The number of errors that can consecutively occur
+   * before a warmup subscription for a particular request
+   * is cancelled
+   */
+  unhealthyThreshold: number
+
+  /**
+   * Hashing options for differentiating requests
+   */
+  hashOpts: Required<Parameters<typeof objectHash>>['1']
+}
+
+export function get(): Config {
+  return {
+    hashOpts: {
+      algorithm: 'sha1',
+      encoding: 'hex',
+      excludeKeys: (props: string) =>
+        ['id', 'maxAge', 'meta']
+          .concat((process.env.CACHE_KEY_IGNORED_PROPS || '').split(',').filter((k) => k))
+          .includes(props),
+    },
+    unhealthyThreshold: Number(process.env.WARMUP_UNHEALTHY_THRESHOLD) || 3,
+    warmupInterval: (Number(process.env.CACHE_MAX_AGE) || 30_000) / 2,
```

Comment:
Sure, we can address this when RL gets merged in, I just wanted to try and avoid env var overload for node operators if possible

Reply:
I am now thinking of a production problem and how we handle it. Currently, warmer on/off, and the interval is coupled to cache params which means if we discover a problem we need to disable the cache to disable the warmer. But that wouldn't really work...

Similarly, a separate on/off config or interval, how we use it now, wouldn't work as changes to config would not influence already started streams. We need a circuit breaker. It would be great if we can find a simple solution just to have that option to shut the warmer down in case of emergency.